packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -0,0 +1,707 @@
+
+        if (this._options.color) {
+            uniformBuffer.updateColor3("grl_singleColor", this._options.color ?? Color3.White());
+        }
+
+        uniformBuffer.setTexture("grl_colors", this._colorsTexture ?? GreasedLinePluginMaterial._EmptyColorsTexture);
+    }
+
+    /**
+     * Prepare the defines
+     * @param defines
+     * @param _scene
+     * @param _mesh
+     */
+    prepareDefines(defines: MaterialGreasedLineDefines, _scene: Scene, _mesh: AbstractMesh) {
+        const options = this._options;
+        defines.GREASED_LINE_HAS_COLOR = !!options.color;
+        defines.GREASED_LINE_SIZE_ATTENUATION = options.sizeAttenuation ?? false;
+        defines.GREASED_LINE_COLOR_DISTRIBUTION_TYPE_LINE = options.colorDistributionType === GreasedLineMeshColorDistributionType.COLOR_DISTRIBUTION_TYPE_LINE;
+    }
+
+    /**
+     * Get the class name
+     * @returns class name
+     */
+    getClassName() {
+        return GreasedLinePluginMaterial.GREASED_LINE_MATERIAL_NAME;
+    }
+
+    /**
+     * Get shader code
+     * @param shaderType vertex/fragment
+     * @returns shader code
+     */
+    getCustomCode(shaderType: string): Nullable<{ [pointName: string]: string }> {
+        if (shaderType === "vertex") {
+            return {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                CUSTOM_VERTEX_DEFINITIONS: `
+                    attribute vec4 grl_previousAndSide;
+                    attribute vec4 grl_nextAndCounters;
+                    attribute float grl_widths;
+                    attribute vec3 grl_offsets;
+                    attribute float grl_colorPointers;
+
+                    varying float grlCounters;
+                    varying float grlColorPointer;
+
+                    vec2 grlFix( vec4 i, float aspect ) {
+                        vec2 res = i.xy / i.w;
+                        res.x *= aspect;
+                        return res;
+                    }
+                `,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                CUSTOM_VERTEX_UPDATE_POSITION: `
+                    vec3 grlPositionOffset = grl_offsets;
+                    positionUpdated += grlPositionOffset;
+                `,
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                CUSTOM_VERTEX_MAIN_END: `
+                    vec2 grlResolution = grl_resolution_lineWidth.xy;
+                    float grlBaseWidth = grl_resolution_lineWidth.z;
+
+                    grlColorPointer = grl_colorPointers;
+
+                    vec3 grlPrevious = grl_previousAndSide.xyz;
+                    float grlSide = grl_previousAndSide.w;
+
+                    vec3 grlNext = grl_nextAndCounters.xyz;
+                    grlCounters = grl_nextAndCounters.w;
+
+                    float grlAspect = grlResolution.x / grlResolution.y;

```

Comment:
should we do the divide in the cpu side instead (more perf and less bandwith)

Reply:
Yep! Sure. I'd like to add two more attributes: `grlColorPointersOffset' and `grllOffsetsOffset' for animation support. I'll pack these two with the `grlAspect` into a vec3.